client/src/features/action/needs/NeedFormGoods.tsx

```diff
@@ -55,8 +57,13 @@ const initialFormData: ShareANeedData = {
 
 function NeedForm() {
   const [formData, setFormData] = React.useState<ShareANeedData>(initialFormData);
+  const [formInProgress, setFormInProgress] = React.useState<boolean>(false);
   const history = useHistory();
 
+  React.useEffect(() => {
+    setFormInProgress(() => DetectFormData(formData));
```

Comment:
any specific reason for the arrow function?
or would `setFormInProgress(DetectFormData(formData));` work?

Reply:
I don't ever set state in that manner, it can lead to unintended results in certain scenarios.